registry/registry.go

```diff
@@ -56,7 +71,7 @@ func hasFile(files []os.FileInfo, name string) bool {
 // provided TLS configuration.
 func ReadCertsDirectory(tlsConfig *tls.Config, directory string) error {
 	fs, err := ioutil.ReadDir(directory)
-	if err != nil && !os.IsNotExist(err) {
+	if err != nil && !os.IsNotExist(err) && !os.IsPermission(err) {
```

Comment:
This was the only change I had to give some thinking (could this hide actual errors?) I _guess_ in the "rootful" situation, `root` would always have access, unless (e.g.) a "read-only filesystem", AppArmor, or Seccomp would also show up as a "permission" error (?)

Would be good if someone could try if that's possible, and if so, we should to a follow-up to log the `os.IsNotExist(err)` error as a Warning instead of fully ignoring it.

Reply:
@AkihiroSuda Actually; now that we're using a different path for the rootless situation; why would we want a permission error to be ignored at all? Shouldn't it remain an error situation?

Reply:
@sh7dm ptal?

Reply:
Hmm, I don't really remember why it was added.

Reply:
This was introduced before, so daemon won't fail if it can't find or access `certs.d`, the final fix I did in second commit was searching for `certs.d` in `$XDG_CONFIG_HOME`.

Reply:
Thanks! So looks like we can remove that;

```suggestion
	if err != nil && !os.IsNotExist(err) {
```

I'll open a PR on master

Reply:
:+1: Great! Thanks!

Reply:
Looks like it's a separate commit, @AkihiroSuda, so you can drop the commit from this cherry-pick to address it.

For master, I opened https://github.com/moby/moby/pull/40576, which reverts the commit

Reply:
dropped this commit